presto-main/src/main/java/com/facebook/presto/storage/TempStorageManager.java

```diff
@@ -114,12 +114,18 @@ public TempStorage getTempStorage(String name)
         return tempStorage;
     }
 
+    public void loadTempStorages(Map<String, Map<String, String>> storageProperties)
+    {
+        storageProperties.entrySet().stream()
+                .forEach(entry -> loadTempStorage(entry.getKey(), entry.getValue()));
+    }
+
     protected void loadTempStorage(String name, Map<String, String> properties)
     {
         requireNonNull(name, "name is null");
         requireNonNull(properties, "properties is null");
 
-        log.info("-- Loading temp storage --");
+        log.info("-- Loading temp storage %s --", name);
 
         TempStorageFactory factory = tempStorageFactories.get(name);
```

Comment:
With the current model it is not allowed to register more than a single storage for a single storage factory. We need to refactor it. The usual way it's done in Presto is the following

`String name` parameter set's the name of a created storage. The `Map<String, String> properties` contains a factory name property, say `temp-storage-factory.name=local`. We extract that property and select a factory based on it. Then we remove the property from the map, and supply the remaning properties to the factory to create a storage. Then we register the storage based on the `String name` supplied. You can check how it is done for catalogs: https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/metadata/StaticCatalogStore.java#L114